lib/resources/host.rb

```diff
@@ -138,6 +140,26 @@ def missing_requirements(_protocol)
       # be enumerated in a skip_resource message
       []
     end
+
+    def resolve_with_dig(hostname)
```

Comment:
Yeah I can totally do that. You thinking DarwinHostProvider and LinuxHostProvider would subclass from it, and we'd move both `resolve_with_bind` and `resolve_with_getent` into it?